[Radian Krisno] iP

radiankrisno opened the pull request:
# Dub 😺 
>"Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Dub frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/radiankrisno/ip/releases/tag/v0.1)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉 

And it is **FREE**!
Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

radiankrisno committed:
Finish Level-1, create parser and processor class

radiankrisno committed:
Finish Level-2, create TaskList and Task class

radiankrisno committed:
Finish Level-3, enable mark as done

radiankrisno committed:
Finish level-4, create ToDo, Event, Deadline class

radiankrisno committed:
Finish A-TextUiTesting, add input.txt and update EXPECTED.TXT

radiankrisno committed:
Finish Level-5, handle errors

radiankrisno committed:
Finish Level-6, implement delete feature

radiankrisno committed:
Change package name into all lower case

radiankrisno committed:
Restructure project

radiankrisno committed:
Add save functionality to the hard disk

radiankrisno committed:
Merge pull request #1 from radiankrisno/branch-Level-7

Add save functionality to the hard disk

radiankrisno committed:
Add feature that makes the chatbot understands date and time

radiankrisno committed:
Merge pull request #2 from radiankrisno/branch-Level-8

Add feature that makes the chatbot understands date and time

radiankrisno committed:
Refactor code to add more OOP

radiankrisno committed:
Merge pull request #3 from radiankrisno/branch-A-MoreOOP

Refactor code to add more OOP

radiankrisno committed:
Add JUnit test for all of the class

radiankrisno committed:
Merge pull request #4 from radiankrisno/branch-A-JUnit

Add JUnit test for all of the class

radiankrisno committed:
Publish Dub jar

radiankrisno committed:
Implement find feature which allows user to find task based on the keyword

radiankrisno committed:
Add javadoc for all of the methods

radiankrisno committed:
Change boolean naming to follow coding standard

radiankrisno committed:
Change switch statement to follow coding standard

radiankrisno committed:
Merge pull request #5 from radiankrisno/branch-A-JavaDoc

Add javadoc for all of the methods

radiankrisno committed:
Merge master branch

radiankrisno committed:
Merge pull request #6 from radiankrisno/branch-A-CodingStandard

Branch a coding standard

radiankrisno committed:
Merge master

radiankrisno committed:
Merge pull request #7 from radiankrisno/branch-Level-9

Branch level 9

radiankrisno committed:
Merge pull request #8 from radiankrisno/add-gradle-support

Add Gradle support

radiankrisno committed:
Merge pull request #9 from radiankrisno/branch-A-Gradle

Branch a gradle

radiankrisno committed:
Adds checkstyle.xml and suppressions.xml

radiankrisno committed:
Fix code to follow coding convention for main file and exception package

radiankrisno committed:
Fix code to follow coding convention for models package

radiankrisno committed:
Fix code to follow coding convention for parser, processor, and storage package

radiankrisno committed:
Fix code to follow coding convention for tasklist and ui package

radiankrisno committed:
Fix code to follow coding convention for all test

radiankrisno committed:
Merge pull request #10 from radiankrisno/branch-A-CheckStyle

Branch a check style

radiankrisno committed:
Add baseline chatbot gui

radiankrisno committed:
Move Duke.java to duke package

radiankrisno committed:
Update chatbot gui to return the correct response

radiankrisno committed:
Merge pull request #11 from radiankrisno/branch-Level-10

Branch level 10

radiankrisno committed:
User varargs to improve find feature of the chatbot

radiankrisno committed:
Merge pull request #12 from radiankrisno/branch-A-Varargs

User varargs to improve find feature of the chatbot

DystoriaX commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
```
Seems like you renamed `Duke` into `duke.Duke` 😮

DystoriaX commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
    1. If there are any further prompts, accept the defaults.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
This too, might be misleading

radiankrisno replied:
Thank you for the comments! Noted

DystoriaX commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+import processor.Processor;
+import storage.Storage;
+import ui.Ui;
+
```
You may want to add the comment header for this class. See [this](https://se-education.org/guides/conventions/java/intermediate.html#comments).

DystoriaX commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Listen to the parser and user input.
+     */
+    public static void listen() {
+        while (true) {
+            boolean bool = parser.nextLine().getKey();
+            if (!bool) {
+                break;
+            }
+        }
+    }
```
You can use a more descriptive name rather than `bool`.

DystoriaX commented on src/main/java/exception/DukeException.java:
```diff
@@ -0,0 +1,15 @@
+package exception;
+
+/**
+ * Exception class for duke.Duke processes.
```
You may want to consider use 'Duke' instead of 'duke.Duke'.

DystoriaX commented on src/main/java/gui/DialogBox.java:
```diff
@@ -0,0 +1,62 @@
+
+/**
+ * An example of a custom control using FXML.
+ * This control represents a dialog box consisting of an ImageView to represent the speaker's face and a label
```
Even though it's exactly 110 in length (which still complies with the coding standard), perhaps would be a good idea to break it into two lines. It's too long to be seen when people review the code.

radiankrisno replied:
Got it!

DystoriaX commented on src/main/java/gui/MainWindow.java:
```diff
@@ -0,0 +1,53 @@
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+/**
```
Seems better if you put one blank line after the `import`

DystoriaX commented on src/main/java/models/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+     */
+    @Override
+    public String toString() {
+        if (this.isDone) {
+            return "[D][X] " + this.description + " (by: "
+                    + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+        }
+        return "[D][ ] " + this.description + " (by: "
+                + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
Might be repetitive if you return the 'almost' the same string representation. You may want to store a variable called `iconString` which contains `"[X]"` or `"[ ]"` based on `this.isDone`. Then you can concatenate the strings altogether. Seems neater to me this way.

DystoriaX commented on src/main/java/models/Event.java:
```diff
@@ -0,0 +1,59 @@
+     * @return String implementation of the Event.
+     */
+    @Override
+    public String toString() {
+        if (this.isDone) {
+            return "[E][X] " + this.description + " (by: "
+                    + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+        }
+        return "[E][ ] " + this.description + " (by: "
+                + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
```
This too

DystoriaX commented on src/main/java/models/Task.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Make the Task done.
```
Makes

DystoriaX commented on src/main/java/models/Task.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Equals function that check whether 2 objects are equal or not.
```
I'd prefer the method header starts with singular verb. See [this](https://se-education.org/guides/conventions/java/intermediate.html#comments).

So you can say: "Returns true if two objects are equal."

DystoriaX commented on src/main/java/models/Task.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Return string implementation of the Task object.
```
Returns

DystoriaX commented on src/main/java/models/Todo.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    /**
+     * Equals function that check whether 2 objects are equal or not.
```
This too

DystoriaX commented on src/main/java/models/Todo.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    /**
+     * Return string implementation of the Todo object.
```
Returns

DystoriaX commented on src/main/java/processor/Processor.java:
```diff
@@ -0,0 +1,182 @@
+                LocalDate time = LocalDate.parse(input[1].trim());
+                this.storage.addTask(new Event(input[0], time));
+            } else {
+                throw new DukeException("I don't understand:(");
```
I guess would be better if you add space before the emoticon `:(`

radiankrisno replied:
Updated!

DystoriaX commented on src/main/java/processor/Processor.java:
```diff
@@ -0,0 +1,182 @@
+    }
+
+    /**
+     * Process command implementation with type LIST.
```
I'd prefer "Processes LIST type command" as the comment header. Not sure whether we actually allowed to use nouns.

DystoriaX commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,156 @@
+    }
+
+    /**
+     * Retrieve the TaskList saved inside the local storage file.
```
Retrieves

DystoriaX reviewed (commented):
In general, nice code, LGTM. Some mistakes that are common is the comment headers for methods. Seems like it should start with singular verbs instead of nouns.

radiankrisno committed:
Handle bye input to close gui

radiankrisno committed:
Fix coding style

hernpiblo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    /**
+     * Listen to the parser and user input.
```
Javadoc can use "Listens" instead of "Listen". Minor issue but seen in a number of places

rldefa replied:
Javadoc should use singular verb.

hernpiblo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    /**
+     * Return the response of the chatbot according to the user input.
```
Same issue, "Returns" instead of "Return"

hernpiblo commented on src/main/java/models/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+ */
+public class Deadline extends Task {
+
+    /** The deadline of the task. */
```
I like how you added Javadocs for private fields also :)

hernpiblo reviewed (commented):
Overall very few coding standard violations, except for those that was mentioned.
Also, I think you can add gradle files to gitignore, if im not wrong.
Other than that that looks good to me

rldefa commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|";
+        Ui.print("Hello from\n" + logo);
+        Ui.print("Wait, I'm not duke.Duke. I'm Dub!\nWhat can I do for you?");
```
Nice.

rldefa reviewed (commented):
Good and clean code. Very nice.

radiankrisno committed:
Fix MainWindow getResponse function call

radiankrisno committed:
Update build.gradle main class

After retructuring the folder, the main class of build.gradle is
pointing into seedu.duke.Duke.

This commit change the main class for the jar file to point back
at duke.Duke, which is the correct main class.

radiankrisno committed:
Update build.gradle main class to show gui

Previously, build.gradle main class is pointing to the main function
for the CLI application.

This commit makes the main class in build.gradle to point to Launcher
main function.

radiankrisno committed:
Enable assertions and check tasks added are not null

radiankrisno committed:
Merge pull request #13 from radiankrisno/branch-A-Assertions

Enable assertions and check tasks added are not null

radiankrisno committed:
Avoid magic number in parser

radiankrisno committed:
Minimize code duplication by using variable for time in string format

radiankrisno committed:
Merge pull request #14 from radiankrisno/branch-A-CodeQuality

Branch a code quality

radiankrisno committed:
Add gradle.yml

radiankrisno committed:
Update gradle.yml

radiankrisno committed:
Merge pull request #15 from radiankrisno/branch-A-CI

Add gradle.yml

radiankrisno committed:
Implement undo and redo feature

radiankrisno committed:
Update test cases for TaskList

radiankrisno committed:
Merge pull request #16 from radiankrisno/branch-C-Undo

Implement undo and redo feature

radiankrisno committed:
Add icon for the application

radiankrisno committed:
Add circle clip to display pictures

radiankrisno committed:
Improve chatbox UI

radiankrisno committed:
Merge pull request #18 from radiankrisno/branch-A-BetterGui

Branch a better gui

radiankrisno committed:
Apply undo feature to setDone function

radiankrisno committed:
Edit undo and redo function to write to local memory

radiankrisno committed:
Add Ui.png

radiankrisno committed:
Add javadoc comments for gui package

radiankrisno committed:
Add javadoc comments for storage and tasklist package

radiankrisno committed:
Add javadoc comments for processor package

radiankrisno committed:
Add new exception

radiankrisno committed:
Follow checkstyle

radiankrisno committed:
Rename to Ui.png

radiankrisno committed:
Add new InvalidIndexException

radiankrisno committed:
Add user guide

radiankrisno committed:
Update user guide

radiankrisno committed:
Set theme jekyll-theme-midnight

radiankrisno committed:
Trigger rebuild

radiankrisno committed:
Change theme to hacker

radiankrisno committed:
Update user guide

radiankrisno committed:
Repair ordered list in user guide

radiankrisno committed:
Move picture in user guide

radiankrisno committed:
Undo moving picture in user guide

radiankrisno committed:
Update javadoc commands to follow style
